js/client-battle.js

```diff
@@ -690,6 +691,18 @@
 		},
 		toggleIgnoreSpects: function(e) {
 			this.battle.ignoreSpects = !!e.currentTarget.checked;
+			this.battle.add('Spectators ' + (this.battle.ignoreSpects ? '' : 'no longer ') + 'ignored.');
+		},
+		toggleIgnoreOpponent: function(e) {
+			if (!!e.currentTarget.checked) {
+				app.ignore[toUserid(this.battle.yourSide.name)] = 1;
```

Comment:
I'm not sure if this is the right implementation. It's not clearly too clearly communicated that the user is ignored globally and not just in the battle. We might want to ignore the user just in the battle.